packages/profile-sync-controller/src/controllers/user-storage/schema.ts

```diff
@@ -3,28 +3,40 @@ import { createSHA256Hash } from './encryption';
 /**
  * The User Storage Endpoint requires a feature name and a namespace key.
  * Developers can provide additional features and keys by extending these types below.
+ *
+ * Adding ALLOW_ARBITRARY_KEYS as the first key in the array allows for any key to be used for this feature.
+ * This can be useful for features where keys are not deterministic (eg. accounts addresses).
  */
+const ALLOW_ARBITRARY_KEYS = 'ALLOW_ARBITRARY_KEYS' as const;
 
 export const USER_STORAGE_SCHEMA = {
   notifications: ['notificationSettings'],
+  accounts: [ALLOW_ARBITRARY_KEYS],
```

Comment:
Like it! This makes it expressive that it expects any arbitrary key (rather than an empty or null)